app/demo-loop/Transporter.cc

```diff
@@ -290,18 +298,20 @@ TransporterResult Transporter<M>::operator()(const TrackInitParams& primaries)
 
         result.time.steps.push_back(get_step_time());
 
-        if (--remaining_steps == 0)
+        if (CELER_UNLIKELY(--remaining_steps == 0))
         {
-            // Exceeded step count
+            CELER_LOG(warning) << "Exceeded step count of " << input_.max_steps;
```

Comment:
Definitely got bit in the ass a few times not realizing I had reached `max_steps`, so this is a wise addition!

Reply:
Perhaps I should upgrade the message to an error since it means the results will be incorrect?

Reply:
Might not be a bad idea. Another thought is to make it so that not specifying `max_steps` just defaults to making it an infinite loop (until no tracks are alive). So, if you really ask for a step limitation, you probably had a good reason to do so (e.g., doing a quick run to get some measurements).